ids not encoded in response from `GET /api/quotas`

**Describe the bug**
For example:

```
[
      {
            "model_class": "Quota",
            "id": 1,
            "name": "Test",
            "quota_source_label": null,
            "url": "*deprecated attribute, URL not filled in by server*"
      }
]
```

Found by BioBlend CI tests: https://github.com/galaxyproject/bioblend/actions/runs/7535163574/job/20510698403#step:10:3567

```
self = <bioblend._tests.TestGalaxyQuotas.TestGalaxyQuotas testMethod=test_get_quotas>

    def test_get_quotas(self):
        quotas = self.gi.quotas.get_quotas()
>       assert self.quota["id"] in [quota["id"] for quota in quotas]
E       AssertionError: assert '1cd8e2f6b131e891' in [3]

bioblend/_tests/TestGalaxyQuotas.py:28: AssertionError
```

Probably introduced in https://github.com/galaxyproject/galaxy/pull/17262 .

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: dev
Commit: 7af682d76f54317e3ca33bcde4dd01639ec4458a

**To Reproduce**
Steps to reproduce the behavior:
1. Create a quota from the admin interface
2. Visit <galaxy_url>/api/quotas while logged in as an admin

**Expected behavior**
Ids should be encoded, but are not.

--- a/lib/galaxy/quota/_schema.py
+++ b/lib/galaxy/quota/_schema.py
@@ -11,7 +11,7 @@ from pydantic import (
 from typing_extensions import Literal
 
 from galaxy.schema.fields import (
-    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
     ModelClassField,
 )
 from galaxy.schema.schema import (
@@ -107,7 +107,7 @@ class QuotaBase(Model, WithModelClass):
     """Base model containing common fields for Quotas."""
 
     model_class: QUOTA = ModelClassField(QUOTA)
-    id: DecodedDatabaseIdField = Field(
+    id: EncodedDatabaseIdField = Field(
         ...,
         title="ID",
         description="The `encoded identifier` of the quota.",

--- a/lib/galaxy/webapps/galaxy/services/quotas.py
+++ b/lib/galaxy/webapps/galaxy/services/quotas.py
@@ -55,13 +55,13 @@ class QuotasService(ServiceBase):
             encoded_id = Security.security.encode_id(quota.id)
             item["url"] = url_for(route, id=encoded_id)
             rval.append(item)
-        return QuotaSummaryList.model_construct(root=rval)
+        return QuotaSummaryList(root=rval)
 
     def show(self, trans: ProvidesUserContext, id: DecodedDatabaseIdField, deleted: bool = False) -> QuotaDetails:
         """Displays information about a quota."""
         quota = self.quota_manager.get_quota(trans, id, deleted=deleted)
         rval = quota.to_dict(view="element", value_mapper={"total_disk_usage": float})
-        return QuotaDetails.model_construct(**rval)
+        return QuotaDetails(**rval)
 
     def create(self, trans: ProvidesUserContext, params: CreateQuotaParams) -> CreateQuotaResult:
         """Creates a new quota."""
@@ -71,7 +71,7 @@ class QuotasService(ServiceBase):
         item = quota.to_dict()
         item["url"] = url_for("quota", id=Security.security.encode_id(quota.id))
         item["message"] = message
-        return CreateQuotaResult.model_construct(**item)
+        return CreateQuotaResult(**item)
 
     def update(self, trans: ProvidesUserContext, id: DecodedDatabaseIdField, params: UpdateQuotaParams) -> str:
         """Modifies a quota."""

--- a/test/integration/test_quota.py
+++ b/test/integration/test_quota.py
@@ -25,6 +25,18 @@ class TestQuotaIntegration(integration_util.IntegrationTestCase):
         json_response = index_response.json()
         assert len(json_response) > 0
 
+    def test_index_returns_encoded_ids(self):
+        quota = self._create_quota_with_name("test-index-encoded-quota")
+        created_quota_id = quota["id"]
+        index_response = self._get("quotas")
+        index_response.raise_for_status()
+        json_response = index_response.json()
+        assert len(json_response) > 0
+        quota_ids = [quota["id"] for quota in json_response]
+        for quota_id in quota_ids:
+            assert isinstance(quota_id, str)
+        assert created_quota_id in quota_ids
+
     def test_index_deleted(self):
         quota = self._create_quota_with_name("test-index-deleted-quota")
         quota_id = quota["id"]

